File: temp/testMap.js
// SIMPLE TEST MAP
var testMap = {

  height: 70,
  width: 70,
  filler: "PLIN",

  players:[

    { name:"Player 1", gold:1000 , team:1,
      units:[
        { x:2, y:2, type:"TANK"    , ammo:1, hp:60, owner:0, fuel:5 },
        { x:4, y:4, type:"APCR"    , ammo:5, hp:100, owner:0, fuel:25 },
        { x:5, y:2, type:"INFT_OS" , ammo:0, hp:100, owner:0, fuel:10 },
        { x:2, y:9, type:"TANK"    , ammo:5, hp:50, owner:0, fuel:50 }
      ]
    },

    { name:"Player 2", gold:0    , team:2,
      units:[
        { x:6, y:10, type:"INFT_OS" , ammo:0, hp:67, owner:1, fuel:50 },
        { x:3, y:14, type:"INFT_OS" , ammo:0, hp:29, owner:1, fuel:5 },
        { x:12, y:7, type:"TANK"    , ammo:5, hp:100, owner:1, fuel:50 },
        { x:11, y:9, type:"TANK"    , ammo:5, hp:100, owner:1, fuel:4 },
        { x:7, y:6, type:"TANK"     , ammo:5, hp:100, owner:1, fuel:15 }
      ]
    },

    { name:"Player 3 (allied)", gold:0    , team:1,
      units:[ { x:3, y:4, type:"TANK", ammo:6, hp:72, owner:2, fuel:50 } ]
    }
  ],

  data:{
    "0":{ "4":"FRST",
      "7":"MNTN" },
    "1":{ "4":"FRST",
      "6":"FRST",
      "7":"FRST" },
    "2":{ "3":"FRST",
      "4":"MNTN",
      "7":"FRST" },
    "8":{ 
      "8":"MNTN" },
    "9":{
      "7":"MNTN",
      "8":"FRST" }
  },

  properties:[
    { x:5, y:3, type:"SILO",    capturePoints:0 },
    { x:0, y:5, type:"HQ",      capturePoints:20, owner:0 },
    { x:0, y:6, type:"CITY",    capturePoints:20, owner:0 },
    { x:1, y:5, type:"FACTORY", capturePoints:20, owner:0 },
    { x:8, y:7, type:"HQ",      capturePoints:20, owner:1 },
    { x:9, y:5, type:"FACTORY", capturePoints:20, owner:1 },
    { x:9, y:6, type:"CITY",    capturePoints:20, owner:1 }
  ]
};

File: src/com/engine/TestMap.java
package com.engine;

import com.engine.DetachedMapUtil.MapPlayer;
import com.engine.DetachedMapUtil.PlayerUnit;

/**
 * TestMap.java
 *
 * This class was made so I can make my own test maps to test the engine with.
 *
 * @author Carr, Crecen
 * @license Look into "LICENSE" file for further information
 * @version 12.27.12
 */

public class TestMap {

	public static String createTestMap(){
		DetachedMapUtil util = new DetachedMapUtil();

		//UNCOMMENT HERE TO TEST THE MAP FILE
        util.mapHeight = 10;
        util.mapWidht = 10;

        util.filler = "PLIN";

        util.tiles[5][2] = "MNTN";
        
        util.tiles[6][2] = "HQ";
        util.tiles[9][5] = "FRST";
        util.tiles[1][3] = "MNTN";
        util.tiles[1][4] = "MNTN";
        util.tiles[2][7] = "FRST";
        util.tiles[2][8] = "FRST";
        util.tiles[2][9] = "FRST";

        PlayerUnit unit;
        MapPlayer p1 = new MapPlayer();
        p1.name = "P1";
        p1.gold = 2000;
        p1.team = 1;
        util.players.add(p1);

        unit = new PlayerUnit();
        unit.ammo = 1;
        unit.fuel = 10;
        unit.hp = 30;
        unit.x = 0;
        unit.y = 0;
        unit.type = "INFT_OS";
        p1.units.add(unit);

        unit = new PlayerUnit();
        unit.ammo = -1;
        unit.fuel = 40;
        unit.hp = 50;
        unit.x = 1;
        unit.y = 1;
        unit.type = "INFT_OS";
        p1.units.add(unit);


        MapPlayer p2 = new MapPlayer();
        p2.name = "P2";
        p2.gold = 12000;
        p2.team = 2;
        util.players.add(p2);

        unit = new PlayerUnit();
        unit.ammo = 1;
        unit.fuel = 10;
        unit.hp = 30;
        unit.x = 3;
        unit.y = 4;
        unit.type = "INFT_OS";
        p2.units.add(unit);
        
        return DetachedMapUtil.toJSON(util);
	}
}
